### PR TITLE
week 1 milestones

### PR DESCRIPTION
**Pull request by stmcpeters**
this PR is checking for week 1 milestones

- creating a new React app using Vite
- routes to fetch data from API and database
- organized project structure of components
- adding simple components to verify that app is running correctly

Please review the app structure and suggest any improvements

### CODE REVIEW

**Comment by stmcpeters** [comment]
hey @derekhouck, since our meeting, I was able to display details about the random movie (see commit comments for more details). With this, I believe I achieved all of this week's milestones but if you could review what I have so far and suggest any improvements, that'd be appreciated!
Screenshot 2024-10-18 at 2 36 15 PM

File: client/src/components/MovieRec.jsx [hunk 1]
```diff
@@ -5,23 +5,21 @@ import React from 'react'
...
-  //   const randomIndex = Math.floor(Math.random() * movies.length);
-  //   return movies[randomIndex];
-  // }
+  const selectRandomMovie = (movies) => {
+    const randomIndex = Math.floor(Math.random() * movies.length);
+    return movies[randomIndex];
+  }
 
-  // const randomMovie = selectRandomMovie(movies);
+  const randomMovie = selectRandomMovie(movies);
```
Comment (derekhouck) [inline client/src/components/MovieRec.jsx]: I think your rendering issues come from here. This should be moved to a React `useState` hook.

**Review by derekhouck — APPROVED** [review]
I'm approving so you can merge this in, but I'd like to see another PR that updates `App.jsx` to only render the `MovieRec` component after the `movies` data has loaded.

**Comment by stmcpeters** [comment]
thanks for the very helpful feedback! I was able to:
- implement useState to set and update randomMovie 
- used useEffect() to run selectRandomMovie() if/when the movies array has been properly loaded
- create a loading message for the users, to display if the movies array has not properly loaded yet

(see commit `4f2da57` for more details)

**Comment by stmcpeters** [comment]
as an added bonus, I was also able to figure out react-router-dom. I have set up pages for user authentication (user login or sign up) as the initial page presented to users and a "home" page for users to select genres and be shown the movie details and reviews. next steps are to conditionally route to certain pages. for example, users are automatically routed to "home" page after either signing up or logging in

**Pull request merged**